### PR TITLE
add debcheck script for repoclosure like analyzis

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: scripts/debcheck.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  focal)
+    # this is broken, as ubuntu needs multiple archives to be enabled to work
+    DEBIAN_PACKAGES_SOURCE=http://archive.ubuntu.com/ubuntu/dists/${DEBIAN_RELEASE}/main/binary-amd64/Packages.xz
+    ;;
+  *)
+    DEBIAN_PACKAGES_SOURCE=https://deb.debian.org/debian/dists/${DEBIAN_RELEASE}/main/binary-amd64/Packages.xz
+    ;;
+esac
+
+PUPPET_PACKAGES_SOURCE=https://apt.puppet.com/dists/${DEBIAN_RELEASE}/puppet6/binary-amd64/Packages
```
Comment (ekohl) [inline scripts/debcheck.sh:21]: Should we default to Puppet 7?

**Review by ekohl — APPROVED** [review]

File: scripts/debcheck.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+FOREMAN_PACKAGES_SOURCE=https://deb.theforeman.org/dists/${DEBIAN_RELEASE}/${FOREMAN_VERSION}/binary-amd64/Packages
+PLUGINS_PACKAGES_SOURCE=https://deb.theforeman.org/dists/plugins/${FOREMAN_VERSION}/binary-amd64/Packages
+
+WORKDIR=$(mktemp -d)
+PACKAGES_DEBIAN=${WORKDIR}/Packages_debian
+PACKAGES_PUPPET=${WORKDIR}/Packages_puppet
+PACKAGES_FOREMAN=${WORKDIR}/Packages_foreman
+PACKAGES_PLUGINS=${WORKDIR}/Packages_plugins
```
Comment (ekohl) [inline scripts/debcheck.sh:29]: Should plugins be optional so you can also check core? Any plans to integrate this into CI pipelines?

File: scripts/debcheck.sh (hunk 1 above)
Comment (evgeni) [reply scripts/debcheck.sh:21]: technically, we still support 6, so testing it as the lower boundary is more correct?

File: scripts/debcheck.sh (hunk 2 above)
Comment (evgeni) [reply scripts/debcheck.sh:29]: slow down, will ya? :)

yes, and yes, and I still need to figure out how to properly do this for ubuntu

File: scripts/debcheck.sh (hunk 1 above)
Comment (ekohl) [reply scripts/debcheck.sh:21]: I guess. I also considered a an env var for just the version that allows you to easily test with both. Maybe it's overkill for now.

Comment (evgeni) [reply scripts/debcheck.sh]: that's actually a good idea, done

File: scripts/debcheck.sh (hunk 2 above)
Comment (evgeni) [reply scripts/debcheck.sh:32]: 1. done
2. will be somewhere else ;)

**Review by ekohl — APPROVED** [review]
Looks like a good start. I was thinking about reducing some duplication by using more arrays but declarative array may just complicate things. We can always refactor it later on.

The only thing I'd question is if we need an extension. We have a shebang and if calling this script becomes the "interface", dropping the extension would allow rewriting it in a different language.

File: scripts/debcheck.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
```
Comment (ekohl) [inline scripts/debcheck.sh:1]: Should it use `set -e` to fail early?
```suggestion
#!/bin/bash

set -e
```

Comment (evgeni) [reply scripts/debcheck.sh:1]: and also pipefail, yes!

**Comment by evgeni** [comment]
> The only thing I'd question is if we need an extension. We have a shebang and if calling this script becomes the "interface", dropping the extension would allow rewriting it in a different language.

Stop using my arguments against me!

**Pull request merged**